Dockerfile

```diff
@@ -1,12 +1,55 @@
-FROM ubuntu:latest
+FROM ubuntu:xenial
 
-RUN apt update && apt install -y build-essential git cmake libssl-dev libsodium-dev wget pkg-config
+RUN apt update && apt install -y build-essential curl git cmake libssl-dev libsodium-dev wget pkg-config autoconf libtool
 WORKDIR /usr/src/app
 
-COPY install-deps-linux.sh install-deps-linux.sh
-RUN ./install-deps-linux.sh
-COPY . .
-RUN mkdir -p build && cd build && sodium_LIBRARY_RELEASE="deps/sodium/lib" cmake .. -DBOOST_ROOT="/usr/src/app/deps/boost" -DOPENSSL_ROOT_DIR="/usr/include/openssl/" && cmake --build .
+## Boost
+ARG BOOST_VERSION=1_70_0
+ARG BOOST_VERSION_DOT=1.70.0
+ARG BOOST_HASH=430ae8354789de4fd19ee52f3b1f739e1fba576f0aded0897c3c2bc00fb38778
+RUN set -ex \
+    && curl -s -L -o  boost_${BOOST_VERSION}.tar.bz2 https://dl.bintray.com/boostorg/release/${BOOST_VERSION_DOT}/source/boost_${BOOST_VERSION}.tar.bz2 \
+    && echo "${BOOST_HASH}  boost_${BOOST_VERSION}.tar.bz2" | sha256sum -c \
+    && tar -xvf boost_${BOOST_VERSION}.tar.bz2 \
+    && cd boost_${BOOST_VERSION} \
+    && ./bootstrap.sh \
+    && ./b2 --build-type=minimal link=static runtime-link=static --with-chrono --with-date_time --with-filesystem --with-program_options --with-regex --with-serialization --with-system --with-thread --with-locale threading=multi threadapi=pthread cflags="-fPIC" cxxstd=14 cxxflags="-fPIC" stage
+ENV BOOST_ROOT /usr/local/boost_${BOOST_VERSION}
```

Comment:
Not needed

Reply:
Going to merge now, we can improve it in future PRs.